apps/vaporgui/VaporTable.cpp

```diff
@@ -483,47 +452,35 @@ void VaporTable::GetValues(std::vector<double> &vec)
     }
 }
 
-void VaporTable::highlightActiveRow(int row)
+void VaporTable::_correctImmutableCellText()
 {
-    if (row < 0) return;
-
     for (int i = 0; i < _table->rowCount(); i++) {
+        if ((i == _activeRow) && (_highlightFlags == ROWS)) { continue; }
         for (int j = 0; j < _table->columnCount(); j++) {
-            QWidget *  cell = _table->cellWidget(i, j);
-            QLineEdit *le = qobject_cast<QLineEdit *>(cell);
-            if (le) {
-                if (i == row)
-                    le->setStyleSheet("QLineEdit " + selectionColor);
-                else
-                    le->setStyleSheet("QLineEdit " + normalColor);
-            } else {
-                if (i == row)
-                    cell->setStyleSheet("QWidget " + selectionColor);
-                else
-                    cell->setStyleSheet("QWidget " + normalColor);
-            }
+            QBrush b(QColor("black"));
+            QBrush w(QColor("white"));
+            _table->item(i, j)->setForeground(b);
+            _table->item(i, j)->setBackground(w);
         }
     }
 }
 
-void VaporTable::highlightActiveCol(int col)
+void VaporTable::highlightActiveRow(int row)
 {
-    if (col < 0) return;
+    if (row < 0) return;
 
     for (int i = 0; i < _table->rowCount(); i++) {
         for (int j = 0; j < _table->columnCount(); j++) {
-            QWidget *  cell = _table->cellWidget(i, j);
-            QLineEdit *le = qobject_cast<QLineEdit *>(cell);
-            if (le) {
-                if (j == col)
-                    le->setStyleSheet("QLineEdit " + selectionColor);
-                else
-                    le->setStyleSheet("QLineEdit " + normalColor);
+            if (i == row) {
+                QBrush b(QColor("blue"));
+                QBrush w(QColor("white"));
+                _table->item(i, j)->setForeground(w);
+                _table->item(i, j)->setBackground(b);
             } else {
-                if (j == col)
-                    cell->setStyleSheet("QWidget " + selectionColor);
-                else
-                    cell->setStyleSheet("QWidget " + normalColor);
+                QBrush b(QColor("black"));
+                QBrush w(QColor("white"));
+                _table->item(i, j)->setForeground(b);
+                _table->item(i, j)->setBackground(w);
```

Comment:
You're correct.  This is a custom theme for immutable table cells.

Normally, table cells can be edited by the user, which isn't normally what we want.  When the cells are flagged as non-editable, the text becomes grey.  This is why the black/white foreground and background are being set here.

Reply:
To your above comment (I wish github was better at responding to top-level comments)

> You said you removed column selection because we wouldn’t use it for renderer selection but I think VaporTable is used for other things too. I don’t think we are using the column selection anywhere but I thought I’d point that out.

Part of this PR was making sure it works with the PythonVariables code, which doesn't use column selection.  You're correct, and the Python engine should also be a part of our test suite before release.

Reply:
> When the cells are flagged as non-editable, the text becomes grey.

I don't believe that is the case for the default QTableWidget.

Reply:
Here's a demo.  Untar the files, run qmake, make, then run the application.  The item in column 3 has been flagged as non-editable, and is grayed out.

![Screen Shot 2021-03-18 at 8 27 41 AM](https://user-images.githubusercontent.com/9522770/111642581-d2cdb080-87c3-11eb-93a3-6af88f0351fc.png)

[grayCellDemo.zip](https://github.com/NCAR/VAPOR/files/6164818/grayCellDemo.zip)

Reply:
@sgpearse Is the graying out of non-editable text a new feature? I've noticed, for example, that the grayed out text in that Statistics tool is so faint that it is barely readable.

Reply:
It's not a new feature.

Qt grays out text when you make a table cell non-editable.  Statistics was written before VaporTable, so it doesn't use it.  It will continue to have hard-to-see gray cells until it's updated.